internal/langserver/handlers/did_change_watched_files.go

```diff
@@ -0,0 +1,49 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+
+	ilsp "github.com/hashicorp/terraform-ls/internal/lsp"
+	"github.com/hashicorp/terraform-ls/internal/protocol"
+	lsp "github.com/hashicorp/terraform-ls/internal/protocol"
+	"github.com/hashicorp/terraform-ls/internal/terraform/ast"
+)
+
+func (svc *service) DidChangeWatchedFiles(ctx context.Context, params lsp.DidChangeWatchedFilesParams) error {
+	for _, change := range params.Changes {
+		URI := string(change.URI)
+		dir, err := os.Stat(URI)
+		if err != nil {
+			return fmt.Errorf("Unable to update %s. Failed to open directory, error: %w", URI, err)
+		}
+
+		filename := path.Base(URI)
+		if !dir.IsDir() && !ast.IsModuleFilename(filename) {
+			// TODO: raise warning; unsupported file
+			return fmt.Errorf("Unable to update %s, filetype not supported.", filename)
+		}
+
+		// only handle Changed event type
+		if change.Type == protocol.Changed {
+			dh := ilsp.HandleFromDocumentURI(change.URI)
+
+			// check existence
+			_, err = svc.modStore.ModuleByPath(dh.Dir.Path())
```

Comment:
Based on your comment on the issue page; we're essentially using this to ascertain whether a module associated with the given URI exists or not, right? And since we're not handling events other than Changed type, we're going to do an early return with an error here. Are these assumptions correct? 

https://github.com/hashicorp/terraform-ls/blob/70bb9e5545894cd629a16cf1d3e754fcd9c5d4c1/internal/state/module.go#L352-L354